src/main/groovy/org/labkey/gradle/task/RestartTriggerTask.groovy

```diff
@@ -0,0 +1,43 @@
+package org.labkey.gradle.task
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputDirectory
+import org.labkey.gradle.util.BuildUtils
+
+import java.nio.charset.StandardCharsets
+import java.text.SimpleDateFormat
+
+abstract class RestartTriggerTask extends DefaultTask
+{
+    public static final String RESTART_FILE_NAME = ".restartTrigger"
+    @Input
+    final abstract Property<String> useLocalBuild = project.objects.property(String).convention(project.hasProperty('useLocalBuild') ? (String) project.property('useLocalBuild') : null)
+
+    @OutputDirectory
+    final abstract DirectoryProperty triggerFileDir = BuildUtils.getRootBuildDirectoryProperty(project, "deploy/modules")
+
+    @Input
+    void updateRestartTriggerFile()
+    {
+        if (useLocalBuild.get() == null || "false" == useLocalBuild.get())
+            return
+
+        if (!triggerFileDir.get().asFile.exists())
+            return
+
+        OutputStreamWriter writer = null
+        try {
+            File triggerFile = new File(triggerFileDir.get().asFile, RESTART_FILE_NAME)
+            writer = new OutputStreamWriter(new FileOutputStream(triggerFile), StandardCharsets.UTF_8)
+            writer.write(SimpleDateFormat.getDateTimeInstance().format(new Date()))
+        }
+        finally
+        {
+            if (writer != null)
+                writer.close()
```

Comment:
Just curious: does Groovy have try-with-resources?

Reply:
Last time I tried, it did not, but I can try again.

Reply:
I don't really care, it just seems funny to see this pattern these days

Reply:
Seems like it is now supported. (The lag in language support is certainly one of the reasons I regret having chosen Groovy in the first place. Someday maybe we can make the switch to Java (which Gradle itself did a long time ago).)

Reply:
I would certainly support that change! Could a Groovy --> Java migration happen incrementally? Or is it all or nothing?

Reply:
Pretty sure it's all or nothing, but I've not tried. IntelliJ offers an option to do the work for us, but not sure how well this works. Still, it's very tempting.

![Screenshot 2024-11-04 at 10 03 45 AM](https://github.com/user-attachments/assets/68c0859a-7bde-417b-a0e4-4da4c2f88f2b)